src/script_interface/lees_edwards/LinearShear.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2021-2022 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef SCRIPT_INTERFACE_LEES_EDWARDS_LINEAR_SHEAR_HPP
+#define SCRIPT_INTERFACE_LEES_EDWARDS_LINEAR_SHEAR_HPP
+
+#include "core/lees_edwards.hpp"
+
+#include "script_interface/ScriptInterface.hpp"
+#include "script_interface/auto_parameters/AutoParameters.hpp"
+
+#include <boost/variant.hpp>
+
+#include <memory>
+
+namespace ScriptInterface {
+namespace LeesEdwards {
+
+class LinearShear : public Protocol {
+public:
+  LinearShear()
+      : m_protocol{
+            new ::LeesEdwards::ActiveProtocol{::LeesEdwards::LinearShear()}} {
```

Comment:
For all the shear-cases, is there a reason to use `new` instead of `std::make_shared` for the construction of the core object? Intuitively i would expect it to work out of the box and to be more explicit than this solution.

Reply:
To me it is not clear where this difference comes from, especially because we just construct a shared-pointer from it. I would have expected that `std::make_shared` does exactly that, but I never bothered to check what `std::make_shared` actually does internally. I always had the impression that it is designed to replace `new` with a shared-ownership, which does not seem to be the whole story.

Reply:
The `std::shared_ptr` constructor takes a raw pointer as argument, and takes ownership of it, i.e. we are no longer allowed to call `free()` or `delete` on the original raw pointer. The resulting shared pointer is lightweight, since it only wraps a raw pointer. Whether the pointee is allocated with `new` or `malloc` is not really relevant, but the `new` syntax is a bit more readable. This strategy has some overhead due to the double dereference, but then we can have a lightweight `std::vector<std::shared_ptr<T>>` that is easy to resize and resort.

The `std::shared_ptr` copy constructor is designed to work with `std::make_shared`, which embeds the pointee. For our use case in the script interface, I think it's safe to replace the `new` syntax with the `std::make_shared` everywhere. It also bugged me a few weeks ago when I realized we were using the `new` syntax in the script interface.